From: Michael Chan <michael.chan@broadcom.com>
Date: Sun, 14 Jun 2020 19:57:08 -0400
Subject: bnxt_en: Re-enable SRIOV during resume.
Patch-mainline: v5.8-rc2
Git-commit: 59ae210173ff86256fa0cdba4ea4d608c61e123d
References: jsc#SLE-8371 bsc#1153274

If VFs are enabled, we need to re-configure them during resume because
firmware has been reset while resuming.  Otherwise, the VFs won't
work after resume.

Fixes: c16d4ee0e397 ("bnxt_en: Refactor logic to re-enable SRIOV after firmware reset detected.")
Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/ethernet/broadcom/bnxt/bnxt.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
@@ -12086,6 +12086,8 @@ static int bnxt_resume(struct device *de
 
 resume_exit:
 	bnxt_ulp_start(bp, rc);
+	if (!rc)
+		bnxt_reenable_sriov(bp);
 	rtnl_unlock();
 	return rc;
 }
